rpm/python-cryptography.spec

```diff
@@ -0,0 +1,151 @@
+BuildRequires:  python%{python3_pkgversion}-hypothesis >= 1.11.4
+BuildRequires:  python%{python3_pkgversion}-iso8601
+BuildRequires:  python%{python3_pkgversion}-pretend
+BuildRequires:  python%{python3_pkgversion}-pytest >= 3.2.1
+BuildRequires:  python%{python3_pkgversion}-pytz
+%endif
+%endif
+
+%description
+cryptography is a package designed to expose cryptographic primitives and
+recipes to Python developers.
+
+%if 0%{?with_python2}
+%package -n  python2-%{srcname}
+Summary:        PyCA's cryptography library
+
+%if 0%{?with_python3}
+%{?python_provide:%python_provide python2-%{srcname}}
+%else
+Provides:       python-%{srcname}
+%endif
+
+BuildRequires:       openssl-libs
+BuildRequires:       python2-idna >= 2.1
+BuildRequires:       python2-six >= 1.4.1
+BuildRequires:       python2-cffi >= 1.7
+BuildRequires:       python2-enum34
+BuildRequires:       python2-ipaddress
+
+%description -n python2-%{srcname}
+cryptography is a package designed to expose cryptographic primitives and
+recipes to Python developers.
+%endif
+
+%if 0%{?with_python3}
+%package -n  python%{python3_pkgversion}-%{srcname}
+Summary:        PyCA's cryptography library
+%{?python_provide:%python_provide python%{python3_pkgversion}-%{srcname}}
+
+BuildRequires:       openssl-libs
+BuildRequires:       python%{python3_pkgversion}-idna >= 2.1
+BuildRequires:       python%{python3_pkgversion}-six >= 1.4.1
+BuildRequires:       python%{python3_pkgversion}-cffi >= 1.7
+
+%description -n python%{python3_pkgversion}-%{srcname}
+cryptography is a package designed to expose cryptographic primitives and
+recipes to Python developers.
+%endif
+
+%prep
+%autosetup -p1 -n %{name}-%{version}/upstream
+
+%build
+%if 0%{?with_python2}
+%py2_build
+%endif
+%if 0%{?with_python3}
+%py3_build
+%endif
+
+%install
+# Actually other *.c and *.h are appropriate
+# see https://github.com/pyca/cryptography/issues/1463
+find . -name .keep -print -delete
+
+%if 0%{?with_python2}
+%py2_install
+%endif
+%if 0%{?with_python3}
+%py3_install
+%endif
+
+%check
+%if %{with tests}
+%if 0%{?with_python2}
+# see https://github.com/pyca/cryptography/issues/4885 and
+# see https://bugzilla.redhat.com/show_bug.cgi?id=1761194 for deselected tests
+PYTHONPATH=%{buildroot}%{python2_sitearch} %{__python2} -m pytest -k "not (test_buffer_protocol_alternate_modes or test_dh_parameters_supported or test_load_ecdsa_no_named_curve)"
+%endif
+
+%if 0%{?with_python3}
+PYTHONPATH=%{buildroot}%{python3_sitearch} %{__python3} -m pytest -k "not (test_buffer_protocol_alternate_modes or test_dh_parameters_supported or test_load_ecdsa_no_named_curve)"
+%endif
+%endif
+
+
+%if 0%{?with_python2}
+%files -n python2-%{srcname}
+%doc LICENSE LICENSE.APACHE LICENSE.BSD README.rst docs
+%{python2_sitearch}/%{srcname}
+%{python2_sitearch}/%{srcname}-%{version}-py*.egg-info
+%endif
+
+
+%if 0%{?with_python3}
+%files -n python%{python3_pkgversion}-%{srcname}
+%doc README.rst docs
```

Comment:
I would like to see these moved to doc subpackage similar to what most other projects have or alternatively left out of the package. Also we haven't used %doc for that because it doesn't quite behave the way we wanted (different version numbers in directory names).